src/main/kotlin/org/lsst/dax/albuquery/resources/Async.kt

```diff
@@ -77,37 +79,102 @@ class Async(val metaservDAO: MetaservDAO) {
     lateinit var headers: HttpHeaders
 
     @POST
-    fun createQuery(@QueryParam("query") @FormParam("query") queryParam: String?, postBody: String): Response {
+    fun createQuery(
+        @QueryParam("query")
+        @FormParam("query") queryParam: String?,
+        @QueryParam("RESPONSEFORMAT") @FormParam("RESPONSEFORMAT") formatParam: String?,
+        postBody: String
+    ): Response {
         val query = queryParam ?: postBody
+        val format = formatParam ?: ""
         LOGGER.info("Recieved query [$query]")
-        val mapper: ObjectMapper
+        var mapper: ObjectMapper? = null
         val ct = headers.getRequestHeader(HttpHeaders.ACCEPT).get(0)
-        if (ct == MediaType.APPLICATION_XML)
-            mapper = TableMapper()
-        else // JSON as default
+        if (ct == MediaType.APPLICATION_JSON || format.contains("json"))
             mapper = ObjectMapper().registerModule(KotlinModule())
+        if (mapper == null) // default
+            mapper = TableMapper()
         return createAsyncQuery(metaservDAO, uri, query, mapper, true)
     }
 
     @Timed
     @GET
-    @Path("{id}/results/{result}")
-    @Produces(MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML)
-    fun getQuery(@PathParam("id") queryId: String, @PathParam("result") result: String): Response {
+    @Path("{id}{p: (/parameters)?}")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun getQueryParams(@PathParam("id") queryId: String): Response {
+        val queryTaskFuture = findOutstandingQuery(queryId)
+        if (queryTaskFuture != null) {
+            val queryTask = queryTaskFuture.get(1, TimeUnit.MILLISECONDS)
+            val params = queryTask.phaseInfo.parameters
+            return Response.ok("Query=" + params).build()
+        }
+        else
+            return Response.status(Response.Status.NOT_FOUND).build()
+    }
+
+    @Timed
+    @GET
+    @Path("{id}/error")
+    @Produces(MediaType.APPLICATION_JSON)
+    fun getQueryStatus(@PathParam("id") queryId: String): Response {
+        val queryTaskFuture = findOutstandingQuery(queryId)
+        if (queryTaskFuture != null) {
+            val queryTask = queryTaskFuture.get(1, TimeUnit.MILLISECONDS)
+            val errorFile = queryTask.phaseInfo.error
```

Comment:
errorFile is a confusing variable name to me.  You're getting it out of phaseInfo.error which is a string, so it felt like this is an actual error string.  Now I'm not sure if it's the contents of an error, or a path to an error file?

Reply:
Okay as I keep going it looks like this a string that contains the contents of the error.  I'd probably just name this error instead of errorFile? or perhaps errorFileContents?

Reply:
Updating PhaseInfo.error to PhaseInfo.errorFile - it does contain a file path, where the error content resides.